### ISSUE TITLE
pod update report error 'undefined method `map' for #<String:0x007fbdd9089cd0>'

### ISSUE THREAD

**Issue by antony-liu** [CocoaPods/CocoaPods#3344]
### Command

```
/usr/bin/pod update
```

### Report

* What did you do?

* What did you expect to happen?

* What happened instead?

### Stack

```
   CocoaPods : 0.36.2
        Ruby : ruby 2.0.0p481 (2014-05-08 revision 45883) [universal.x86_64-darwin14]
    RubyGems : 2.4.5
        Host : Mac OS X 10.10.2 (14C1514)
       Xcode : 6.2 (6C131e)
         Git : git version 1.9.5 (Apple Git-50.3)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ d2f263f3a3df0d01742210bcc1f3db18ba7beb0e
```

### Plugins

```
cocoapods-plugins : 0.4.1
cocoapods-trunk   : 0.6.0
cocoapods-try     : 0.4.3
```

### Podfile

```ruby
platform :ios	,  '7.0'
inhibit_all_warnings!
source 'https://github.com/CocoaPods/Specs.git'

target 'innobase' do

    pod 'AFNetworking'
    pod 'KGStatusBar'
    pod 'FMDB'
    pod 'RESideMenu'
    pod 'JSONModel'
    pod 'MBProgressHUD'
    pod 'MWPhotoBrowser'
    pod 'TPKeyboardAvoiding'
    pod 'ALSystemUtilities'
    pod 'UIAlertView-Blocks'
    pod 'Base64nl'
    pod 'MJRefresh'
    pod 'MessageDisplayKit', :git=>'https://github.com/antony-liu/MessageDisplayKit.git'
    pod 'ZXingObjC'
    pod 'RegexKitLite'
end
```

### Error

```
NoMethodError - undefined method `map' for #<String:0x007fbdd9089cd0>
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/executable.rb:57:in `execute_command'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/downloader.rb:11:in `execute_command'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-downloader-0.8.1/lib/cocoapods-downloader/base.rb:152:in `block in executable'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-downloader-0.8.1/lib/cocoapods-downloader/git.rb:65:in `block in clone'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/downloader.rb:44:in `block in ui_sub_action'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/user_interface.rb:49:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/downloader.rb:43:in `ui_sub_action'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-downloader-0.8.1/lib/cocoapods-downloader/git.rb:63:in `clone'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-downloader-0.8.1/lib/cocoapods-downloader/git.rb:29:in `download!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-downloader-0.8.1/lib/cocoapods-downloader/base.rb:80:in `block in download'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/downloader.rb:28:in `block in ui_action'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/user_interface.rb:49:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/downloader.rb:27:in `ui_action'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-downloader-0.8.1/lib/cocoapods-downloader/base.rb:78:in `download'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/external_sources/abstract_external_source.rb:103:in `block in pre_download'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/user_interface.rb:70:in `titled_section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/external_sources/abstract_external_source.rb:99:in `pre_download'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/external_sources/downloader_source.rb:13:in `fetch'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/installer/analyzer.rb:347:in `fetch_external_source'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/installer/analyzer.rb:324:in `block (2 levels) in fetch_external_sources'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/installer/analyzer.rb:323:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/installer/analyzer.rb:323:in `block in fetch_external_sources'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/user_interface.rb:49:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/installer/analyzer.rb:322:in `fetch_external_sources'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/installer/analyzer.rb:57:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/installer.rb:189:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/installer.rb:111:in `block in resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/user_interface.rb:49:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/installer.rb:110:in `resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/installer.rb:90:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/command/project.rb:71:in `run_install_with_update'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/command/project.rb:155:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-0.8.1/lib/claide/command.rb:312:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/lib/cocoapods/command.rb:46:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-0.36.2/bin/pod:44:in `<top (required)>'
/usr/bin/pod:23:in `load'
/usr/bin/pod:23:in `<main>'
```

**Comment by EpaL** [comment]
I just updated to 0.36.2 and got the same error.

**Comment by raven** [comment]
+ 1

**Comment by aidenluo177** [comment]
I just updated to 0.36.2 and got the same error too.

**Comment by jcalvert** [comment]
I was about to file an issue; I just got this by doing `rake pod:install` from a fresh RedPotion created app. I changed my gemfile to 0.36.1 and the problem went away. The code is expecting an array apparently, but simply changing it be an array of strings still results in errors.

**Comment by EpaL** [comment]
Back up and running again:

```
[]$ sudo gem uninstall cocoapods

Select gem to uninstall:
 1. cocoapods-0.28.0
 2. cocoapods-0.34.4
 3. cocoapods-0.35.0
 4. cocoapods-0.36.0.beta.2
 5. cocoapods-0.36.0
 6. cocoapods-0.36.1
 7. cocoapods-0.36.2
 8. All versions
> 7
Successfully uninstalled cocoapods-0.36.2
```

**Comment by antony-liu** [comment]
Uninstall cocoapods-0.36.2 and run 'pod update' using version 0.36.1, this problem went away.

**Comment by ajfigueroa** [comment]
Reverting back a version fixed the issue for me.

**Comment by leafduo** [comment]
Same here

**Comment by segiddins** [comment]
I'll be releasing 0.36.3 with a fix momentarily.

**Comment by crossle** [comment]
How to use master branch? or select 0.36.1?

**Comment by segiddins** [comment]
This should now be fixed in 0.36.3.
